### PR TITLE
bug fixes

### PR DESCRIPTION
**Pull request by identicalsnowflake**
See my comment on the commit itself.

### CODE REVIEW

**Comment by jb55** [comment]
thanks!

File: src/Network/S3/URL.hs [hunk 1]
```diff
@@ -32,14 +32,14 @@ canonicalRequest S3Request{..} =
...
-    seconds    = pack (formatTime defaultTimeLocale "T%M%H%SZ" requestTime)
+    seconds    = pack (formatTime defaultTimeLocale "T%H%M%SZ" requestTime)
     date       = pack (formatTime defaultTimeLocale "%Y%m%d" requestTime)
     timeHeader = s3Header "x-amz-date" (date <> seconds)
     headers    = sortS3Headers (timeHeader : hostHeader : hashHeader : s3headers)
     headerKeys = map (fst . getS3Header) headers
 
     httpMethod       = renderS3Method s3method
-    canonicalURI     = HTTP.urlEncodeBuilder False objectName
+    canonicalURI     = fromByteString objectName
```
Comment (jb55) [inline src/Network/S3/URL.hs]: this looks suspicious to me (because of this: https://docs.aws.amazon.com/AmazonS3/latest/API/images/sigV4-using-query-params.png) see `UrlEncode(resource)`

is this change necessary to make it work for you?

Comment (identicalsnowflake) [reply src/Network/S3/URL.hs]: Well I was storing/retrieving items in places like `123/456.jpg`, and that `/` was being encoded and messing things up (if I just uploaded to `123.jpg`, everything would work). Removing this call to `HTTP.urlEncodeBuilder` made `123/456.jpg` work correctly. Whether that's the proper general solution, I don't know, as my file names are very tame and don't contain any weird characters.

File: src/Network/S3/Sign.hs [hunk 2]
```diff
@@ -31,7 +31,7 @@ hmac_ key = toBytes . hmacGetDigest . hmacSHA256 key
 sign :: ByteString -> S3Request -> UTCTime -> S3SignedRequest
 sign key req utc =
   let
-      seconds   = BS.pack (formatTime defaultTimeLocale "T%M%H%SZ" utc)
+      seconds   = BS.pack (formatTime defaultTimeLocale "T%H%M%SZ" utc)
```
[the remark this answers is not in the document]
Comment (dmjio) [reply src/Network/S3/Sign.hs]: @jb55 bet `H` and `M` are the same in the AWS example, probably why it didn't matter.
